physical/oci_objectstorage/oci_objectstorage_ha.go

```diff
@@ -0,0 +1,554 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package oci_objectStorage
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/oracle/oci-go-sdk/objectstorage"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// The lock implementation below prioritizes ensuring that there are no 2 primary at any given point in time
+// over high availability of the primary instance
+
+// Verify Backend satisfies the correct interfaces
+var _ physical.HABackend = (*Backend)(nil)
+var _ physical.Lock = (*Lock)(nil)
+
+const (
+	// LockRenewInterval is the time to wait between lock renewals.
+	LockRenewInterval = 3 * time.Second
+
+	// LockRetryInterval is the amount of time to wait if the lock fails before trying again.
+	LockRetryInterval = 5 * time.Second
+
+	// LockWatchRetryInterval is the amount of time to wait if a watch fails before trying again.
+	LockWatchRetryInterval = 2 * time.Second
+
+	// LockTTL is the default lock TTL.
+	LockTTL = 15 * time.Second
+
+	// LockWatchRetryMax is the number of times to retry a failed watch before signaling that leadership is lost.
+	LockWatchRetryMax = 4
+
+	// LockCacheMinAcceptableAge is minimum cache age in seconds to determine that its safe for a secondary instance
+	// to acquire lock.
+	LockCacheMinAcceptableAge = 45 * time.Second
+
+	// LockWriteRetriesOnFailures is the number of retries that are made on write 5xx failures.
+	LockWriteRetriesOnFailures = 4
+
+	ObjectStorageCallsReadTimeout = 3 * time.Second
```

Comment:
These timeouts seem quite low?

Reply:
It's fine for us so far since we launch to production last 6 months. What is your suggestion for this?

Reply:
It may be OK, it's just lower than some of the defaults in some HA clients like Dynamo and GC (I'm going from memory). But if customers notice a lot of leadership changes in their monitoring, we might want to revisit it.